File: Assignment 3/C# version/Exercise 6/Point.cs
using System;
namespace Assignment1
{
	public class Point
	{
		private int x;
		private int y;

		public Point()
		{
			x = 0;
			y = 0;
		}

		public Point(int x, int y)
		{
			this.x = x;
			this.y = y;
		}

		public string toString()
		{
			return "(" + x + "," + y + ")";
		}

		public bool isEqualTo(Point p)
		{
			return p.x == x && p.y == y;
		}

		public double distanceTo(Point p)
		{
			return Math.Sqrt(Math.Pow(x - p.x, 2) + Math.Pow(y - p.y, 2));
		}

		public void move(int x, int y)
		{
			this.x += x;
			this.y += y;
		}

		public void moveToXY(int x, int y)
		{
			this.x = x;
			this.y = y;
		}
	}
}

File: Uni/sb223ce_assign3/Point.java
package sb223ce_assign3;

public class Point {

	
	
	private int x; // create field variables
	private int y;

	
	
	
	public Point() { // create empty constructor
	}

	
	
	
	public Point(int a, int b) { // create constructor with parameters

		x = a; // give parameter values to field variables
		y = b;
	}

	
	
	
	public Point(Point p) {   // other Constructor for comparing two Point equation
		x = p.x; /* give new constructor param variables values to field
					      variables**/
		y = p.y;
	}

	
	
	
	public String toString() { // method syntax
		return "(" + x + "," + y + ")"; // return string
	}

	
	
	
	public int getX() { // method syntax
		return x; // return value to field variable
	}

	
	
	
	public int getY() { // method syntax
		return y; // return value to field variable
	}

	
	
	
	public boolean isEqualTo(Point p) { // method syntax

		return x == p.getX() && y == p.getY(); // return true or false
	}

	
	
	
	public double distanceTo(Point p) { // method syntax

		int x2 = p.getX(); // create two integer variable for saving param
							// variable values
		int y2 = p.getY();

		double distance;
		distance = Math.sqrt(Math.pow(x - x2, 2) + Math.pow(y - y2, 2)); // use
																			// distance
																			// formula
		return distance; // return distance 
	}

	
	
	
	public void move(int a, int b) { // method syntax
		x = x + (a); /*add or subtract the value of field variable according to parameter
						 variable values**/
		y = y + (b);
	}// no return, because it is void method

	
	
	
	public void moveToXY(int a, int b) { // method syntax
		x = a; // move the field variables as given param variable value
		y = b;

	}// no return, because it is void method

}
